lib/db/activity.ts

```diff
@@ -1,19 +1,37 @@
+import { auth } from "../auth/auth";
 import { Activity } from "@prisma/client";
 import { ActivityWithIncludes } from "./types";
 import prisma from "./prisma";
 
 /**
  * Creates an activity log entry.
  *
- * @param {Omit<Activity, "id">} params - Activity log details
- * @returns {Promise<Activity>} - Created activity log entry
+ * @param type - Type of action ("Create", "Update", or "Delete")
+ * @param entity - Model that was changed
+ * @param entity_id - ID of the entity that was changed
+ * @returns {Promise<Activity>} - The created activity log entry
  */
 export async function createActivityLog(
-  params: Omit<Activity, "id">
+  type: string,
+  entity: string,
+  entity_id: number,
+  user_id?: number
 ): Promise<Activity> {
+  const session = await auth();
+
+  if (!session || !session.user || !user_id) {
```

Comment:
The `user_id` parameter is optional and unused in your calls, causing this function to throw on every call. Looks like you forgot to remove it here, or maybe you were trying to do `!session.user.id`?